File: src/test/java/database/DatabaseManagerTest.java
package database;

import org.junit.Before;
import org.junit.Test;

import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

public class DatabaseManagerTest {
  private DatabaseManager dbManager;

  @Before
  public void setUp() throws Exception {
    dbManager = DatabaseManager.getInstance(null);
  }

  @Test
  public void testAddDatabase() {
    dbManager.addDatabase(MongoDbManager.getInstance());
    assertTrue(dbManager.hasDatabaseManager(MongoDbManager.getInstance()));
    assertFalse(dbManager.hasDatabaseManager(MySqlManager.getInstance()));
  }

  @Test
  public void testRemoveDatabase() {
    dbManager.addDatabase(MongoDbManager.getInstance());
    assertTrue(dbManager.hasDatabaseManager(MongoDbManager.getInstance()));
    dbManager.removeDatabase(MongoDbManager.getInstance());
    assertFalse(dbManager.hasDatabaseManager(MongoDbManager.getInstance()));
  }
}

File: src/test/java/database/DatabaseManagerTest.kt
package database

import database.DatabaseManager.Companion.getInstance
import database.MongoDbManager.Companion.instance
import org.junit.Assert
import org.junit.Before
import org.junit.Test
import sensormodels.DatabaseModel

class DatabaseManagerTest {
    private var dbManager: DatabaseManager? = null

    @Before
    @Throws(Exception::class)
    fun setUp() {
        dbManager = getInstance(null) as DatabaseManager
    }

    @Test
    fun testAddDatabase() {
        dbManager!!.addDatabase(instance as DbManager<DatabaseModel>)
        Assert.assertTrue(dbManager!!.hasDatabaseManager(instance as DbManager<DatabaseModel>))
    }

    @Test
    fun testRemoveDatabase() {
        dbManager!!.addDatabase(instance as DbManager<DatabaseModel>)
        Assert.assertTrue(dbManager!!.hasDatabaseManager(instance as DbManager<DatabaseModel>))
        dbManager!!.removeDatabase(instance as DbManager<DatabaseModel>)
    }
}